auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -6,17 +6,48 @@
 import type { SubscriptionClient, TenantIdDescription } from '@azure/arm-subscriptions'; // Keep this as `import type` to avoid actually loading the package before necessary
 import type { TokenCredential } from '@azure/core-auth'; // Keep this as `import type` to avoid actually loading the package (at all, this one is dev-only)
 import * as vscode from 'vscode';
+import type { AzureAuthentication } from './AzureAuthentication';
 import type { AzureSubscription, SubscriptionId, TenantId } from './AzureSubscription';
+import type { AzureSubscriptionProvider } from './AzureSubscriptionProvider';
 import { NotSignedInError } from './NotSignedInError';
 import { getConfiguredAuthProviderId, getConfiguredAzureEnv } from './utils/configuredAzureEnv';
-import type { AzureAuthentication } from './AzureAuthentication';
-import type { AzureSubscriptionProvider } from './AzureSubscriptionProvider';
+
+const EventDebounce = 5 * 1000; // 5 seconds
 
 /**
  * A class for obtaining Azure subscription information using VSCode's built-in authentication
  * provider.
  */
-export class VSCodeAzureSubscriptionProvider implements AzureSubscriptionProvider {
+export class VSCodeAzureSubscriptionProvider extends vscode.Disposable implements AzureSubscriptionProvider {
+    private readonly onDidSignInEmitter = new vscode.EventEmitter<void>();
+    private lastSignInEventFired: number = 0;
+
+    private readonly onDidSignOutEmitter = new vscode.EventEmitter<void>();
+    private lastSignOutEventFired: number = 0;
+
+    public constructor() {
+        const disposable = vscode.authentication.onDidChangeSessions(async e => {

```

Comment:
TODO: I don't know for sure, but I think this may fire when `getSubscriptions` is called and new sessions are being created for each tenant. If so we need to suppress firing the event while `getSubscriptions` is running.